### PR TITLE
Rust parse serialized

### PR DESCRIPTION
**Pull request by arvidn**
This is an attempt at speeding up Streamable deserialization of `SerializedProgram`. We need to half-parse the CLVM buffer to know the size of it.

The main part of this change is here: https://github.com/Chia-Network/clvm_rs/pull/77

### CODE REVIEW

**Review by mariano54 — DISMISSED** [review]
Awesome! ill test it on my benchmark

**Comment by arvidn** [comment]
this can't land as-is. The `setup.py` commit needs to be updated and we need to cut a new `clvm_rs` release first too.

File: setup.py [hunk 1]
```diff
@@ -6,7 +6,8 @@
     "chiabip158==1.0",  # bip158-style wallet filters
     "chiapos==1.0.1",  # proof of space
     "clvm==0.9.6",
-    "clvm_rs==0.1.6",
+    "maturin",
+    "clvm_rs@git+https://github.com/Chia-Network/clvm_rs.git@main#egg=clvm-rs",  # noqa
```
Comment (richardkiss) [inline setup.py:10]: We can't do this in something going to pypi, unfortunately we have to do a real release of `clvm_rs`.

**Review by richardkiss — CHANGES_REQUESTED** [review]
Let's create a release of `clvm_rs`.

File: setup.py (hunk 1 above)
Comment (richardkiss) [reply setup.py:10]: Oh, okay, you're aware.

**Review by richardkiss — APPROVED** [review]

File: chia/types/blockchain_format/program.py [hunk 2]
```diff
@@ -148,8 +148,8 @@ class SerializedProgram:
 
     @classmethod
     def parse(cls, f) -> "SerializedProgram":
-        tmp = sexp_buffer_from_stream(f)
-        return SerializedProgram.from_bytes(tmp)
+        length = serialized_length(f.getvalue()[f.tell() :])
```
Comment (richardkiss) [inline chia/types/blockchain_format/program.py:151]: It's possible that casting the bytes into a `memoryview` will save one copy of the blob. Might be worth checking to see how it benchmarks.

```python
length = serialized_length(memoryview(f.getvalue())[f.tell():])
```

(I've been meaning to experiment with `memoryview` in python `clvm` to see if it makes for zero-cost `substr`.)

Comment (arvidn) [reply chia/types/blockchain_format/program.py:151]: yeah, my initial idea was to use `getbuffer()`, but to feel confident with that API I would want to add more tests to the actual python bindings of `clvm_rs` too.

I agree that a longer term solution would be to take buffer views.

**Pull request merged**